core/app/models/spree/stock/estimator.rb

```diff
@@ -15,8 +20,14 @@ class OrderRequired < StandardError; end
       #   descending cost, with the least costly marked "selected"
       def shipping_rates(package, frontend_only = true)
         raise ShipmentRequired if package.shipment.nil?
-        raise OrderRequired if package.shipment.order.nil?
+        raise OrderRequired if package.shipment.order.nil? && @order.nil?
 
+        if order && package.shipment.order.nil?
+          package.shipment.order = order
+        elsif package.shipment.order && order.nil?
+          @order = package.shipment.order
+        end
```

Comment:
What if `package.shipment.order` and `order` are both set but are two different objects? Should we think about that case?

Reply:
@aldesantis  I think that if the `order` is passed into the constructor it would make sense to give it preference over `package.shipment.order`, I noted that decision in the commit here https://github.com/solidusio/solidus/pull/4105/commits/7506823349f011769c882ffb07151586f234738c

Reply:
@HendrikLouw understood, but what if they're both passed and they're different objects? Is that a data integrity issue, is that something we should be concerned about, etc.?

Reply:
@aldesantis I think it makes sense to make the order passed into the initializer the default, the aim is to remove the usage of `package.shipment.order` and forcing the usage of the passed in order promotes that goal. There shouldn't be any data integrity issues if all the packages always contain the same order. 

If the assumption that packages always have the same order holds then the tests document the correct behaviour if there can be different orders per package then this whole PR and issue should be abandoned, since I don't have enough context here I'm assuming that the creator of the issue took this into account.